src/Utils/Hooks/useMatchMedia.ts

```diff
@@ -30,26 +30,34 @@ export function __internal__useMatchMedia(
   mediaQueryString: string,
   { initialMatches = null } = {}
 ) {
-  const [matches, setMatches] = useState(initialMatches)
+  const [matches, setMatches] = useState<boolean | null>(initialMatches)
 
   useEffect(() => {
+    if (typeof window === "undefined" || typeof matchMedia === "undefined") {
+      return
+    }
+
     const mediaQueryList = window.matchMedia(mediaQueryString)
-    // @ts-expect-error PLEASE_FIX_ME_STRICT_NULL_CHECK_MIGRATION
+
     setMatches(mediaQueryList.matches)
-    const handleChange = event => setMatches(event.matches)
 
-    mediaQueryList.addListener(handleChange)
+    const handleChange = (event: MediaQueryListEvent) =>
+      setMatches(event.matches)
+
+    if (mediaQueryList.addEventListener) {
+      mediaQueryList.addEventListener("change", handleChange)
+    } else {
+      mediaQueryList.addListener(handleChange)
+    }
```

Comment:
Annoying in that `addListener` is deprecated and `addEventListener` appears to be not well documented, but the path forward.